File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/CustomerDAO.java
package com.jann_luellmann.thekenapp.data.dao;

import com.jann_luellmann.thekenapp.data.model.Customer;
import com.jann_luellmann.thekenapp.data.model.Drink;

import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Transaction;
import androidx.room.Update;

@Dao
public abstract class CustomerDAO {

    @Query("SELECT * FROM customer")
    public abstract LiveData<List<Customer>> getAll();

    @Query("SELECT * FROM customer WHERE customerId IN (:customerIds)")
    public abstract LiveData<List<Customer>> loadAllByIds(long[] customerIds);

    @Query("SELECT * FROM customer WHERE name LIKE :name LIMIT 1")
    public abstract LiveData<Customer> findByName(String name);

    @Insert
    public abstract long insert(Customer customer);

    @Insert
    public abstract void insertAll(Customer... customers);

    @Update
    public abstract void update(Customer customer);

    @Delete
    public abstract void delete(Customer customer);
}

File: app/src/main/java/com/jann_luellmann/thekenapp/data/dao/CustomerDAO.kt
package com.jann_luellmann.thekenapp.data.dao

import androidx.lifecycle.LiveData
import androidx.room.*
import androidx.room.OnConflictStrategy.IGNORE
import androidx.room.OnConflictStrategy.REPLACE
import com.jann_luellmann.thekenapp.data.model.Customer

@Dao
interface CustomerDAO {
    @get:Query("SELECT * FROM customer")
    val all: LiveData<List<Customer>>

    @Query("SELECT * FROM customer WHERE customerId IN (:customerIds)")
    fun loadAllByIds(customerIds: LongArray): LiveData<List<Customer>>

    @Query("SELECT * FROM customer WHERE cname LIKE :name LIMIT 1")
    fun findByName(name: String): LiveData<Customer>

    @Insert(onConflict = REPLACE)
    fun insert(customer: Customer): Long

    @Insert
    fun insertAll(vararg customers: Customer)

    @Update
    fun update(customer: Customer)

    @Delete
    fun delete(customer: Customer)
}